winml/lib/Api/ImageFeatureValue.h

```diff
@@ -46,9 +46,17 @@ struct ImageFeatureValue : ImageFeatureValueT<ImageFeatureValue, WinML::ILotusVa
   bool IsBatch() { return m_batchSize > 1; }
 
  private:
+  struct TensorAllocatorDeleter {
+    void operator()(OrtAllocator* allocator) noexcept {
+      com_ptr<winmla::IWinMLAdapter> adapter;
+      if (SUCCEEDED(OrtGetWinMLAdapter(adapter.put())))
```

Comment:
>OrtGetWinMLAdapter [](start = 20, length = 18)

would there be a way you can plumb through the adapter here, so we don't have to create it again ?    ImageFeatureValue::GetOrtValue which does the emplace_back has one ....

Reply:
you could just put a member var here of the adapter, keep the vector of allocators, and in the destructor of ImageFeatureValue() loop through the vector calling the adapter to delete them.  #boom

---
In reply to: [356905944](https://github.com/microsoft/onnxruntime/pull/2635#discussion_r356905944) [](ancestors = 356905944)

Reply:
Yes, I thought about it too... I tried a lambda capturing the adapter by copy, but VS was complaining about using a lambda as the deleter of the unique_ptr. Is there a considerable overhead to get the adapter? I thought it would be fine, since ImageFeatureValue::GetOrtValue gets a new one every time.

Reply:
I'll see if I can get a clean way to do it :)